StudioTBA/CoronaIO
# Flock health and infection

## What issue is this PR solving?
Closes: #37 
Closes: #33 

## Description of change
- Horde has health based on its zombie count
- New visual component in HUD to indicate horde's health
- If a horde _kills_ (infects) a civilian, they will turn into a zombie and will follow that same horde

## How to reproduce the changes
1. Go to scene Sandbox
2. See the Health indicator on the HUD
3. Start the scene
4. Add zombies to your horde and notice the health change
5. Split, join hordes and notice the health change
6. Select different hordes using the dropdown and notice the health change
7. Collide with the human
8. See their health change
9. When killing them, see how a new zombie is created and added to the health counter

## Screenshots or GIFs
### Adding zombies to horde
![Adding zombies to horde](https://user-images.githubusercontent.com/10281272/79047567-6d516700-7be5-11ea-9c77-724e1996e5a4.gif)

### Splitting and joining hordes
![Health splitting](https://user-images.githubusercontent.com/10281272/79047571-80fccd80-7be5-11ea-89be-a536477dba61.gif)

### Using the dropdown
![Using dropdown](https://user-images.githubusercontent.com/10281272/79047578-8a863580-7be5-11ea-8b4e-69ae1234831b.gif)

### Infection
![Infection](https://user-images.githubusercontent.com/10281272/79047581-8fe38000-7be5-11ea-87b0-e16802911bc3.gif)
